go/api/go_api/tests/utils.py

```diff
@@ -1,5 +1,27 @@
 from mock import Mock, patch
 
+from vumi.tests.utils import MockHttpServer
```

Comment:
I don't have an answer for the practical question, so that's probably "no, we can't easily use it".

As for the philosophy:
 * We regularly (I'd guess 1-5% of builds based on having to restart things, I haven't measured properly) have timeouts and breakages in tests that use the localhost network.
 * Making sure cleanup happens correctly for the real network case is nontrivial and we have to jump through hoops to pass ports around. I don't consider that to be a clean API at all.
 * Almost all the code in `fake_connection` is machinery to manage and hook into connection state in tests. There is very little overriding of existing Twisted code, and everything that does modify existing behaviour checks that it needs to before doing so.
 * `FakeServer` and `FakeHttpServer` allow testing of things that cannot be tested using real network connection without replacing quite a lot of Twisted's internals. You can simulate a connection timeout, for example. Or accept a connection and then not do anything with it.
 * `fake_connection` is entirely deterministic, so it's possible (although perhaps not as easy as I'd like) to write tests that fail if something doesn't happen instead of waiting for a timeout.

Maybe we should discuss this somewhere other than this PR, though?